functions/src/modules/gmail.functions.ts

```diff
@@ -0,0 +1,273 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import * as Google from 'googleapis';
+import * as querystring from 'querystring';
+import { Message } from 'firebase-functions/lib/providers/pubsub';
+
+const db = admin.database();
+
+export const oauth2init = functions.https.onRequest(
+  async (req: functions.Request, res: functions.Response) => {
+    const { client_key, secret, redirect } = functions.config().gmail;
```

Comment:
Every config value should be documented in `README`.

However I'm wondering whether we can get these values (at least `client_key` and `secret`) automatically from the current Firebase or Google Cloud project? Please research on that, Id' like not to add any unnecessary config parameters.

Reply:
There's no exisiting README.md for the functions. Should I add it?

Reply:
It is in the root README.

Reply:
> If you have a card to research the default configs, you can assign to me.

Do this research as a part of this task please.

Reply:
I spend around 45 minutes reading articles, and did not find a way to use Application Default Credentials for Oauth2. All the examples are showing the `client_key` and `secret` being passed in to the constructor like this. Should we keep looking?

Reply:
No need. Let's get `client_key` and `secret` from config. However the redirect URL, I believe, should be constructed in the code, as it knows the name of the function that should accept the redirect.

Reply:
Just hardcode it like this?

```
    const oauth2Client = new Google.google.auth.OAuth2(
      client_key,
      secret,
     "https://us-central1-audio-seva-team-test.cloudfunctions.net/Gmail-oauth2callback",
    );
```

Or perhaps construct it somehow from the `request` object?